crates/red_knot_python_semantic/src/types.rs

```diff
@@ -570,9 +642,23 @@ impl<'db> Type<'db> {
                 }
                 first_element_truthiness
             }
-            Type::Intersection(_) => {
-                // TODO
-                Truthiness::Ambiguous
+            Type::Intersection(intersection) => {
+                // The truthiness of the intersection is the intersection of the truthiness of its
+                // elements:
+                // - `Ambiguous` ∩ `Truthy` = `Truthy`
+                // - `Ambiguous` ∩ `Falsy` = `Falsy`
+                // - `Truthy` ∩ `Falsy` = `Never`  -- this should be impossible to build
```

Comment:
I'd prefer to use "Python type" notation rather than maths notation in comments... but kudos for knowing the maths notation 😄

```suggestion
                // - `Ambiguous` & `Truthy` == `Truthy`
                // - `Ambiguous` & `Falsy` == `Falsy`
                // - `Truthy` & `Falsy` == `Never`  -- this should be impossible to build
```